README.md

```diff
@@ -10,35 +10,75 @@ This library uses a fluent API that is separated in 3 parts:
+
+![restclient example 1](docs/img/restclient_cat_example_get_option.png?raw=true)
+
+The only thing that changed from the original method is the return type, which changes from `Cat` to `OptionStrict<Cat>`, and we added a line `AddProcessors(new OptionAsNotFoundProcessor<Cat>())`. 
+
+Let's talk about the change in the returning type first. `OptionStrict<T>` is a type from the [Csharp-Monad](https://github.com/louthy/csharp-monad) library, it represents a nullable type (which can be either a value type or reference type). A value with type `OptionStrict<T>` is either empty or has a value. We can use this type to represent our new method: If the API call returns a 200 status code with the JSON body, then we return a `Cat` value; however, if the API call returns a 404 status code, we return an empty value (representd by `Option<Cat>.Nothing`).
+
+To make this possible, we need to introduce processors, as we did with `AddProcessors(new OptionAsNotFoundProcessor<Cat>())`
+
+## Processors
+
+rest-client uses a pipeline of processors that check whether they find a specific HTTP status code, and if they find it they return a value of the returning type.
+
+In the above case, `OptionAsNotFoundProcessor<TResult>` is a processor that returns values of type `OptionStrict<TResult>`. It checks if a response has _404_ status code; if it does it returns `OptionStrict<TResult>.Nothing`; if it doesn't, then it delegates the processing to the rest of the pipeline, which will return a `TResult` value.
+
+In every call there is an implicit processor that returns values of type `TResult`. It checks if the response has a _2xx_ status code, and if it does it deserializes its JSON body into the expected `TResult` type. This processor is the one that is used behind the scenes in all the previous _GET_ examples we've seen.
+
+Without adding a processor like `OptionAsNotFoundProcessor<TResult>` our library would find that the HTTP response has a _404_ status code, and it won't know what to do with it. Thus, processors are a way to tell the library how to handle status code and determine what C# type it must return depending on the status code.
+
+## Managing 4xx / 5xx status codes
+
+With rest-client we can handle an even more complex subject, like managing HTTP errors with C# types. Until know, every 4xx and 5xx error was not known to our program, i.e it was not represented in the return type. Instead, if the API returned such status codes, our program thre and exception and then forgot about it. There exist cases where we want our own program to handle such errors programatically, allowing such errors to alter the control flow of our program. Using `try-catch` blocks would not be appropriate in that case, since try-catch is not the best approach for control flow.
```

Comment:
Monad is just a flavor of our like, you can use this library without using Monad. Every example we present with Monad should show an advantage of usage try-catch vs if or Either.Select